pkg/server/server.go

```diff
@@ -224,6 +224,7 @@ func (s *Server) Run(ctx context.Context) error {
 		apiHandler = WithClusterScope(apiHandler)
 		apiHandler = WithInClusterServiceAccountRequestRewrite(apiHandler, unsafeServiceAccountPreAuth)
 		apiHandler = WithAcceptHeader(apiHandler)
+		apiHandler = WithClusterAnnotation(apiHandler)
```

Comment:
now your code runs too early. It's like shells, called from the outside to inside.